go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -656,3 +724,76 @@ func (oc *Controller) refreshEgressFirewallLogging(egressFirewallNamespace strin
 func buildEgressFwAclName(namespace string, priority int) string {
 	return fmt.Sprintf("egressFirewall_%s_%d", namespace, priority)
 }
+
+func getNodeInternalAddrsToString(node *kapi.Node) []string {
+	v4, v6 := getNodeInternalAddrs(node)
+	var addrs []string
+	for _, addr := range []net.IP{v4, v6} {
+		if addr != nil {
+			addrs = append(addrs, addr.String())
+		}
+	}
+
+	return addrs
+}
+
+func (oc *Controller) updateEgressFirewallForNode(oldNode, newNode *kapi.Node) error {
+
+	var addressesToAdd []string
+	var addressesToRemove []string
+
+	// figure out what operation this is
+	if oldNode != nil && newNode == nil {
+		// delete event
+		addressesToRemove = getNodeInternalAddrsToString(oldNode)
+	} else if oldNode == nil && newNode != nil {
+		// add event
+		addressesToAdd = getNodeInternalAddrsToString(newNode)
+	} else {
+		// update
+		// check if the node has changed
+		if reflect.DeepEqual(oldNode.Labels, newNode.Labels) &&
+			reflect.DeepEqual(oldNode.Status.Addresses, newNode.Status.Addresses) {
+			return nil
+		}
+		addressesToRemove = getNodeInternalAddrsToString(oldNode)
+		addressesToAdd = getNodeInternalAddrsToString(newNode)
+	}
+
+	// cycle through egress firewalls and check if any match this node's labels
+	var efErr error
+	oc.egressFirewalls.Range(func(k, v interface{}) bool {
+		ef := v.(*egressFirewall)
+		namespace := k.(string)
+		ef.Lock()
+		defer ef.Unlock()
+		for _, rule := range ef.egressRules {
+			// nodeSelector will always have a value, but it is mutually exclusive from cidrSelector and dnsName
+			if len(rule.to.cidrSelector) != 0 || len(rule.to.dnsName) != 0 {
+				continue
+			}
+			selector, err := metav1.LabelSelectorAsSelector(&rule.to.nodeSelector)
+			if err != nil {
+				klog.Errorf("Error while parsing label selector %#v for egress firewall in namespace %s",
+					rule.to.nodeSelector, namespace)
+				continue
+			}
+			// no need to check selector on old node here, ips are unique and regardless of if selector
+			// matches or not we shouldn't have those addresses anymore
+			rule.to.nodeAddrs.Delete(addressesToRemove...)
+			// check if selector matches
+			if selector.Matches(labels.Set(newNode.Labels)) {
+				rule.to.nodeAddrs.Insert(addressesToAdd...)
+			}
```

Comment:
I think it will be better to only update one ACL here instead of updating all ACLs for ef.
I can do that in my next PR, since otherwise this rebase will get more painful

Reply:
done